app/experimenter/static/js/components/DesignForm.js

```diff
@@ -86,6 +89,7 @@ class DesignForm extends React.PureComponent {
       .catch(err => {
         this.setState({
           errors: err.data,
+          saving: false,
```

Comment:
i think this works in practice because we redirect the user on success but if we didn't this would not re-enable the button after successful saves. a better approach may be:
```js
requestSave
  .then(...)
  .catch(...)
  .finally(() => {
    this.setState({
      saving: false,
    });
  })
```

`finally` is always called whether the promise resolves successfully or not.

Docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

Reply:
ok @rehandalal, I think I successfully made this change.